android/src/main/kotlin/com/instrumentisto/medea_flutter_webrtc/media/MediaDevices.kt

```diff
@@ -221,6 +270,19 @@ class MediaDevices(val state: State, val permissions: Permissions) : BroadcastRe
    * [MediaDevices] has ongoing request).
    */
   private suspend fun stopBluetoothSco() {
+    if (Build.VERSION.SDK_INT >= 31) {
+      // Prefer modern routing API on Android 12+
+      audioManager.clearCommunicationDevice()
+      stopBluetoothScoDeferred?.complete(Unit)
+      stopBluetoothScoDeferred = null
+      bluetoothScoDeferred?.completeExceptionally(
+          GetUserMediaException(
+              "Bluetooth headset connection request was cancelled",
+              GetUserMediaException.Kind.Audio))
+      bluetoothScoDeferred = null
+      return
+    }
+
     stopBluetoothScoDeferred = CompletableDeferred()
     audioManager.stopBluetoothSco()
```

Comment:
Some code duplication here and there but it will go away in the next PR where I'll add some kind of a `AudioManagerCompat`